src/client/pps/pps.proto

```diff
@@ -687,6 +687,34 @@ message RunCronRequest {
   Pipeline pipeline = 1;
 }
 
+message CreateSecretRequest {
+  bytes file = 1;
+  string namespace = 2;
+}
+
+message DeleteSecretRequest {
+  string secret  = 1;
+  string namespace = 2;
+}
+
+message InspectSecretRequest {
+  string secret = 1;
+  string namespace = 2;
+}
+
+message ListSecretRequest {
+  string namespace = 1;
+}
+
+message SecretInfo {
+  string name = 1;
+  string type = 2;
+  google.protobuf.Timestamp creation_timestamp = 3;
+}
+
+message SecretInfos {
+  repeated SecretInfo secret_info = 1;
+}
```

Comment:
To make this a little more inline with our other methods I think it'd be good to add a `Secret` message which is used to identify secrets, like this:

```
message Secret {
    string name = 1;
    string namespace = 2;
}
```

and then use that message in `CreateSecretRequest`, `DeleteSecretRequest` `InspectSecretRequest` and `SecretInfo`.

Reply:
Actually thinking about this a little more, is there any reason you should be able to create secrets in different namespaces than the one that pachd is deployed in? I don't think we give people the option to create pipelines outside of that namespace, so I think you can simplify a bit and remove the namespace field and just have that be inherited from the deployment. The pps api server has a field `namespace` in the struct that will contain the namespace it's currently deployed in.